File: wordcount/perl/src/WordCountReducer.pl
#!/usr/bin/perl
use strict;
use warnings;

my %result;
while (defined(my $line = <STDIN>)) {
    chomp($line);
    # ワード毎に加算
    $result{$line}++;
}

# 結果を出力
foreach my $key (keys(%result)){
    print "$key\t$result{$key}\n";
}

File: wordcount/php/src/WordCountReducer.php
#!/usr/bin/php
<?php

function reduce($lines)
{
    $result = array();
    while (!feof($lines)) {
        $line = trim(fgets($lines));
        if (!empty($line)) {
            // キー(aaaとか)の個数をカウント
            $result[$line]++;
        }
    }

    // 結果出力
    foreach ($result as $key => $val) {
        echo "${key}\t${val}\n";
    }
}

reduce(STDIN);
